src/libmongoc/src/mongoc/mongoc-cursor.c

```diff
@@ -1449,9 +1469,16 @@ void
 mongoc_cursor_set_batch_size (mongoc_cursor_t *cursor, uint32_t batch_size)
 {
    BSON_ASSERT (cursor);
+   bson_iter_t iter;
 
-   _mongoc_cursor_set_opt_int64 (
-      cursor, MONGOC_CURSOR_BATCH_SIZE, (int64_t) batch_size);
+   if (bson_iter_init_find (&iter, &cursor->opts, MONGOC_CURSOR_BATCH_SIZE) &&
```

Comment:
The error handling of `mongoc_cursor_set_batch_size` appears to be underspecified. I would like to set `cursor->error` if `_mongoc_cursor_set_opt_*()` returns `false` (currently ignored), but given that would be an API break (a previously never-erroring function now potentially setting an error), perhaps may be worth adding a `MONGOC_WARNING` on failure to set the option instead.

Reply:
Related to / superceded by https://github.com/mongodb/mongo-c-driver/pull/1300/files#r1224707455.

> I agree using `MONGOC_WARNING` seems more appropriate.